### ISSUE TITLE
Failure to start after installing in VS2017

### ISSUE THREAD

**Issue by mpalandri** [BIDeveloperExtensions/bideveloperextensions#43]
VS 2017 Pro 
Version 15.8.5 VisualStudio.15.Release/15.8.5+28010.2036
Microsoft .NET Framework Version 4.7.02556

After installation, nothing from the extension appeared to have loaded.  In Tools|Options|BIDS Helper|Version, get the following error information:

BIDS Helper encountered an error when Visual Studio started:
BIDS Helper encountered an error when Visual Studio started:
Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at BIDSHelper.BIDSHelperPackage.Initialize() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\BidsHelperPackage.cs:line 120
Could not load file or assembly 'Microsoft.AnalysisServices.Common.FrontEnd, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.
Fusion Log:
WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

Could not load file or assembly 'Microsoft.AnalysisServices.Common.FrontEnd, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.
Fusion Log:
WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

Could not load file or assembly 'Microsoft.DataTransformationServices.VsIntegration, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.
Fusion Log:
WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

Could not load file or assembly 'Microsoft.AnalysisServices.Common.FrontEnd, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.
Fusion Log:
WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

BIDS Helper encountered an error when Visual Studio started:
BIDS Helper plugin constructor failed on TabularHideMemberIfPlugin: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BIDSHelper.BIDSHelperPackage.Initialize() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\BidsHelperPackage.cs:line 154

Inner exception:
Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BIDSHelper.BIDSHelperPackage.Initialize() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\BidsHelperPackage.cs:line 154
Inner exception:
Could not load file or assembly 'Microsoft.AnalysisServices.Common.FrontEnd, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.
   at BIDSHelper.TabularHideMemberIfPlugin..ctor(BIDSHelperPackage package)BIDS Helper encountered an error when Visual Studio started:
BIDS Helper plugin constructor failed on TriStatePerspectivesPlugin: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BIDSHelper.BIDSHelperPackage.Initialize() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\BidsHelperPackage.cs:line 154

Inner exception:
Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BIDSHelper.BIDSHelperPackage.Initialize() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\BidsHelperPackage.cs:line 154
Inner exception:
Could not load file or assembly 'Microsoft.AnalysisServices.Design, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.
   at BIDSHelper.TriStatePerspectivesPlugin.OnWindowActivated(Window GotFocus, Window LostFocus)
   at BIDSHelper.BIDSHelperWindowActivatedPluginBase.OnEnable() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\Core\BIDSHelperWindowActivatedPluginBase.cs:line 104
   at BIDSHelper.Core.BIDSHelperPluginBase..ctor(BIDSHelperPackage package) in C:\projects\BIDS Helper\GitHub\bideveloperextensions\Core\BIDSHelperPluginBase.cs:line 45
   at BIDSHelper.TriStatePerspectivesPlugin..ctor(BIDSHelperPackage package) in C:\projects\BIDS Helper\GitHub\bideveloperextensions\SSAS\TriStatePerspectivesPlugin.cs:line 24BIDS Helper encountered an error when Visual Studio started:
BIDS Helper plugin constructor failed on TabularSyncDescriptionsPlugin: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BIDSHelper.BIDSHelperPackage.Initialize() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\BidsHelperPackage.cs:line 154

Inner exception:
Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at BIDSHelper.BIDSHelperPackage.Initialize() in C:\projects\BIDS Helper\GitHub\bideveloperextensions\BidsHelperPackage.cs:line 154
Inner exception:
Could not load file or assembly 'Microsoft.AnalysisServices.Common.FrontEnd, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.
   at BIDSHelper.SSAS.TabularSyncDescriptionsPlugin..ctor(BIDSHelperPackage package)

**Comment by furmangg** [comment]
Did you install SSDT or do you just have the base Visual Studio installation?
https://docs.microsoft.com/en-us/sql/ssdt/download-sql-server-data-tools-ssdt?view=sql-server-2017

**Comment by mpalandri** [comment]
Yes, I have SSDT installed, v 15.1.61808.07020 

Mike 
.

**Comment by furmangg** [comment]
When you installed SSDT did you install Analysis Services projects, Reporting Services projects or Integration Services projects. One of those is required. Or did you just install the database tools part?

Look I'm Help... About Microsoft Visual Studio and copy that info here please.

**Comment by Jandho** [comment]
I have the same problem as mpalandri after upgrading yesterday to 15.8.6

My installation:

Microsoft SQL Server Data Tools for Visual Studio 2017  (SSDT)
Version 15.8.6
VisualStudio.15.Release/15.8.6+28010.2041
Microsoft .NET Framework
Version 4.7.03056

Installed Version: IDE Standard

Microsoft Visual Studio Tools for Applications 2017   00371-40000-00001-AA414
Microsoft Visual Studio Tools for Applications 2017

BI Developer Extensions 2017 Extension   2.1.1
BI Developer Extensions 2017 Visual Studio Extension v2.1.1 - An add-in to extend SQL Server Data Tools

C# Tools   2.9.0-beta8-63208-01
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

NuGet Package Manager   4.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

Project   1.0
Information about my package

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

SQL Server Analysis Services   15.0.900.163
Microsoft SQL Server Analysis Services Designer 
Version 15.0.900.163

SQL Server Data Tools   15.1.61808.07020
Microsoft SQL Server Data Tools

SQL Server Integration Services   
Microsoft SQL Server Integration Services Designer
Version 15.0.900.30

SQL Server Reporting Services   15.0.900.163
Microsoft SQL Server Reporting Services Designers 
Version 15.0.900.163

SsmsVsIntegration   1.0
Information about my package

TFS Source Control Explorer Extension   1.0
Visual Studio Extension for Team Foundation Server Source Control Explorer

Visual Basic Tools   2.9.0-beta8-63208-01
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

**Comment by mpalandri** [comment]
I installed Integration Services. Getting the package development environment was the reason I installed SSDT. 

Mike 
.

**Comment by dgosbell** [comment]
@mpalandri  this might be a little tricky to figure out as we don't get this error on our machines.
 
Is this a relatively new install of Visual Studio 2017? 

Do you have a local install of SQL Server on this machine or just the client tools?

Taking the first error from your initial post:

> Could not load file or assembly 'Microsoft.AnalysisServices.Common.FrontEnd, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91' or one of its dependencies. The system cannot find the file specified.

When I search my c: drive I found 2 versions of this assembly:
v15 in the following folder:
`C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\SSAS`

and v14 in the following folder:
`C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\PrivateAssemblies\Business Intelligence Semantic Model`

Can you check these 2 locations on your system and let us know if you have one or both of these files?

I'm speculating that maybe you are missing the v14 version and this might be the cause of this issue as the current version of BI Developer Extensions was compiled against the v14 version.

**Comment by mpalandri** [comment]
Hi Darren, 

Microsoft.AnalysisServices.Common.frontend.dll exists only in this folder: 
C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\ManagementStudio 

VS 2017 was installed after a complete Windows 10 reinstall last January. 

SQL Server 2016 is installed as well. 

Components installed: 

Microsoft SQL Server 2016 Setup Discovery Report Product Instance Instance ID Feature Language Language Version Clustered Configured 
Microsoft SQL Server 2016 	MSSQLSERVER 	MSSQL13.MSSQLSERVER 	Database Engine Services 	1033 	Developer Edition 	13.1.4001.0 	No 	Yes 
Microsoft SQL Server 2016 	MSSQLSERVER 	MSSQL13.MSSQLSERVER 	SQL Server Replication 	1033 	Developer Edition 	13.1.4001.0 	No 	Yes 
Microsoft SQL Server 2016 	MSSQLSERVER 	MSRS13.MSSQLSERVER 	Reporting Services - Native 	1033 	Developer Edition 	13.1.4001.0 	No 	Yes 
Microsoft SQL Server 2016 			Integration Services 	1033 	Developer Edition 	13.1.4001.0 	No 	Yes 
Microsoft SQL Server 2016 			LocalDB 	1033 	Express Edition 	13.1.4001.0 	No 	Yes 

Mike 
.

**Comment by wmorris20** [comment]
It looks like my issue my be a duplicate of this one.  When I got to see the BIDS Helper Options, I see the same error listed here.

**Comment by wsuhoey** [comment]
I came here about to submit my own issue, but this seems to echo the same error I'm getting.

I'm also getting this same error messages going to Tools | Options | BIDS Helper | Version.

I believe this issue is preventing BIDeveloperExtensions to properly load fully, therefore also preventing me from using the SSIS Deploy Package feature (Package Deployment Model) described on the docs https://bideveloperextensions.github.io/features/DeploySSISPackages/.

I don't have the "Deploy" right-click context menu when I right click a solution or project that's in the package deployment model. I also don't have the "Deploy (BIDS Helper)" properties page.

Above the error text is additional info that hasn't been mentioned yet:

```SSDTBI (SQL Unknown 15.0.1000.59) (15.0.1000.59) for Visual Studio 2017 was detected```
Perhaps it's that "Unknown" part that's throwing it off?

I'm on VS 2017 Pro 15.8.7.
SSDT v15.1.61808.07020 (aka release number 15.8.1, aka build number 14.0.16179.0, released 9/27/2018)
I also have SQL 2017 Developer Edition, SSMS v17.9 installed on this workstation.

```
Microsoft Visual Studio Professional 2017 
Version 15.8.7
VisualStudio.15.Release/15.8.7+28010.2046
Microsoft .NET Framework
Version 4.7.03056

Installed Version: Professional

Visual C++ 2017   00369-60000-00001-AA874
Microsoft Visual C++ 2017

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

Application Insights Tools for Visual Studio Package   8.13.10627.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2017   15.8.05085.0
ASP.NET and Web Tools 2017

ASP.NET Core Razor Language Services   15.8.31590
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2017   5.2.60618.0
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   15.8.05023.0
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.3.5001.1
Microsoft Azure Data Lake Tools for Visual Studio

Azure Stream Analytics Tools for Visual Studio   2.3.5001.1
Microsoft Azure Stream Analytics Tools for Visual Studio

BI Developer Extensions 2017 Extension   2.1.1
BI Developer Extensions 2017 Visual Studio Extension v2.1.1 - An add-in to extend SQL Server Data Tools

C# Tools   2.9.0-beta8-63208-01
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Cookiecutter   15.8.18241.1
Provides tools for finding, instantiating and customizing templates in cookiecutter format.

Find Changeset By Comment   1.0
This extension is used to find changesets that contains a specific comment.

GitHub.VisualStudio   2.5.8.5165
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript Language Service   2.0
JavaScript Language Service

Microsoft Azure HDInsight Azure Node   2.3.5001.1
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.3.5001.1
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.3.5001.1
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2017 - v2.9.10730.2

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   1.0
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   4.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

Office Developer Tools for Visual Studio 2017 ENU   15.0.27907.01
Microsoft Office Developer Tools for Visual Studio 2017 ENU

Productivity Power Tools 2017/2019   16.0
Installs the individual extensions of Productivity Power Tools 2017/2019

Project   1.0
Information about my package

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Python   15.8.18241.1
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.

Python - Django support   15.8.18241.1
Provides templates and integration for the Django web framework.

Python - IronPython support   15.8.18241.1
Provides templates and integration for IronPython-based projects.

Python - Profiling support   15.8.18241.1
Profiling support for Python projects.

Python - UWP support   15.8.18241.1
Provides templates and integration for the UWP framework.

R Tools for Visual Studio   1.3.40517.1016
Provides project system, R Interactive window, plotting, and more for the R programming language.

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

SQL Server Analysis Services   15.0.1000.59
Microsoft SQL Server Analysis Services Designer 
Version 15.0.1000.59

SQL Server Data Tools   15.1.61808.07020
Microsoft SQL Server Data Tools

SQL Server Integration Services   
Microsoft SQL Server Integration Services Designer
Version 15.0.900.30

SQL Server Reporting Services   15.0.900.163
Microsoft SQL Server Reporting Services Designers 
Version 15.0.900.163

SsmsVsIntegration   1.0
Information about my package

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   15.8.20822.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   2.9.0-beta8-63208-01
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.2 for F# 4.5   15.8.0.0.  Commit Hash: 55a3dc3231c95c77f81ee53f7d29152029da7408.
Microsoft Visual F# Tools 10.2 for F# 4.5

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Universal Windows Apps   15.0.28010.2046
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.

Workflow Manager Tools 1.0   1.0
This package contains the necessary Visual Studio integration components for Workflow Manager.```

**Comment by aj311rules** [comment]
I am experiencing the same error. I am running:
Visual Studio 2017 15.8.7
Microsoft SQL Server Data Tools - 15.1.61808.07020 (all 4 components)
SQL Server Management Studio v17.9


Microsoft.AnalysisServices.Common.frontend.dll version 15 exists in C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\CommonExtensions\Microsoft\SSAS

Microsoft.AnalysisServices.Common.frontend.dll version 14 exists in C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\ManagementStudio

**Comment by wmorris20** [comment]
I rolled back successfully to this setup: 

•	Visual Studio 2017 v15.7.6
•	SSDT 2017 v15.1.6
•	BI Developer (BIDs Helper) v2.1.1

**Comment by MaxBarj** [comment]
Hi @wmorris20 

How would you downgrade the version of VS 2017 to v15.7.6 ?

**Comment by wmorris20** [comment]
@MaxBarj

I uninstalled (with cleanup) and reinstalled the specific version.

**Comment by MaxBarj** [comment]
Thanks @wmorris20 
Is it possible to choose the specific version via Visual Studio Installer ? Or did you find the installer for v15.7.6 ?

**Comment by wmorris20** [comment]
@MaxBarj

I believe you have to grab a specific download file:
https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2017-relnotes-v15.7

**Comment by MaxBarj** [comment]
Well the link only provide the changelog for v15.7, all the download links redirect to the visual studio installer.

With a quick Google search I managed to find this useful article:
https://docs.microsoft.com/en-us/visualstudio/productinfo/installing-an-earlier-release-of-vs2017

Thanks for the help, I'll let you know if that solve my problem too.

**Comment by wsuhoey** [comment]
Ideally, we shouldn't have to downgrade our Visual Studio and SSDT. Hopefully the tool's authors can find a permanent solution to this.

**Comment by MaxBarj** [comment]
I have reinstalled VS 2017 15.7.6 with SSDT 15.8.0.
I confirm the extension is now working fine, at least for the SSIS part as I have not tested SSAS and SSRS.

However I still see this weird error on the version checker of BIDS Helper:
![image](https://user-images.githubusercontent.com/44023220/48146733-f1e3db80-e2b5-11e8-8642-a2df5b6e7161.png)

**Comment by furmangg** [comment]
The latest SSDT on VS2017 has upgraded all its DLLs to SQL2019 versions which broke BI Developer Extensions. We are making progress getting BI Developer Extensions upgraded and will keep this thread posted. Thanks for your patience.

**Comment by furmangg** [comment]
Thank you for your patience! We have released an updated version of BI Developer Extensions for Visual Studio 2017 which should resolve all the issues in this thread. Please go to Tools... Extensions and Updates and make sure you are running BI Developer Extensions 2.2.0. It has been tested with the latest SSDT version (version 15.8.2) and 15.8.0. It should also work on older versions of SSDT for Visual Studio 2017.

To recap what happened, the 15.8.x release of SSDT moved to SQL 2019 (v15) version DLLs and broke BI Developer Extensions. But now we have released a fix which supports this.

Please open a new issues if you experience any issues with particular features. Thanks for everyone's effort reporting this and for your patience.
